### PR TITLE
MMT-1816 Added backend changes to properly save deleted data contacts…

### PR DESCRIPTION
**Pull request by ryanmiller-1**
….  Added the ability to delete the last data contact in the ui.

### CODE REVIEW

File: app/assets/javascripts/drafts.coffee [hunk 1]
```diff
@@ -158,20 +166,51 @@ $(document).ready ->
...
     multipleItem = $(this).closest('.multiple-item')
-    $(multipleItem).remove()
+    if !multipleItem.siblings('.multiple-item').length
+      # Reset the form state to original page load.
+      # Adding another and deleting causes the formatting to break, so this is
+      # clearing out the existing fields instead.
+      # In order to return the document to the original state, the document needs
+      # to clear the Country/RelatedURLs before they would be cleared by changing
+      # the contact type, so go up the document in reverse
+      $(multipleItem.find('select:not([disabled="disabled"])').get().reverse()).each (index, element) ->
```
Comment (jceaser) [inline app/assets/javascripts/drafts.coffee]: so we needed to do the reverse after all?

Comment (ryanmiller-1) [reply app/assets/javascripts/drafts.coffee]: Different reverse.  There were two when I demoed it.  The second one was unnecessary, but this one remains so.

File: spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
```
Comment (charleshuang80) [inline spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb]: we don't need this line, it is included in `.rspec`

File: spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            click_on 'Done'
+          end
+        end
+
+        it 'saves the data center' do
+          expect(page).to have_content('AARHUS-HYDRO')
+        end
+
+        it 'saves the removal of the data contact' do
+          expect(page).not_to have_content('Little Johnny Drop Tables')
```
Comment (charleshuang80) [inline spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb]: for consistency purposes this should be `expect(page).to have_no_content('...`

File: spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        # it are working properly.
+        it 'resets variable input boxes' do
+          within '.addresses' do
+            expect(page).to have_css('.state-province-text-field')
+            expect(page).to have_no_css('.state-province-select')
+          end
+        end
+
+        it 'does not have an error message' do
+          expect(page).not_to have_content('This draft has the following errors')
```
Comment (charleshuang80) [inline spec/features/collection_drafts/forms/data_contacts_form/data_contacts_form_delete_spec.rb]: same comment about using `have_no_content`

**Pull request merged**